lohica-js-network/src/components/inputComponents/FormComponent.js

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import {Form, Row, Grid, Button, Alert, Well} from 'react-bootstrap';
+import {connect} from 'react-redux';
+
+import {regValidHandle, regValidHideMes, regValidShowMes, successReg} from '../../actions/inputAction'
+import {signupRoute} from '../../routes';
+
+import {EmailInput, GenderRadio, AgeInput, ImageInput, NameInput, SurnameInput, MidnameInput} from '../index';
+
+class formComponent extends React.Component{
+    constructor(props){
+        super(props);
+
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleSubmit(e){
+        let {emailValid, imageValid, nameValid, surnameValid, genderValidStat, name, surname, midName, email, gender, age} = this.props.store.formInput.regForm;
+        e.preventDefault();
+        if(emailValid && imageValid && nameValid && surnameValid && genderValidStat) {
+
+            let data = new FormData();
+
+            data.append('file', this.props.inputState.regForm.imageData);
+            data.append('name', name);
+            data.append('surname', surname);
+            data.append('midname', midName);
+            data.append('email', email);
+            data.append('gender', gender);
+            data.append('age', age);
+
+            // beginning of fetch
+
+            fetch(signupRoute,{
```

Comment:
server communication in component

Reply:
changed

Reply:
I do not see what you've changed. You still have server communication inside the component.

Reply:
done that